### PR TITLE
Make relative humidity a QuantityAttribute.

### LINKED ISSUE #7665: Unknown ufunc apio13
Travis tests on a [package derived from the astropy template](https://github.com/desihub/specsim) are failing for `ASTROPY_VERSION=development` with multiple instances of:
```
Unknown ufunc apio13.  Please raise issue on https://github.com/astropy/astropy
```
All other combinations in the test matrix are passing.  A sample travis log is [here](https://travis-ci.org/desihub/specsim/jobs/405690907).

### PR DESCRIPTION
**Pull request by mhvk**
This ensures that one can pass in a dimensionless quantity without getting problems with the erfa ufuncs, and also stops being passed in a quantity with a wrong unit (which is why I labeled this as a bug that should be backported). For backwards compatibility, passing in regular numbers remains possible; they are just converted to dimensionless.

fixes #7665

### CODE REVIEW

File: astropy/coordinates/attributes.py [hunk 1]
```diff
@@ -306,7 +306,7 @@ def convert_input(self, value):
         if np.all(value == 0) and self.unit is not None:
             return u.Quantity(np.zeros(self.shape), self.unit), True
         else:
-            if not hasattr(value, 'unit'):
+            if not hasattr(value, 'unit') and self.unit != u.dimensionless_unscaled:
```
Comment (mhvk) [inline astropy/coordinates/attributes.py]: This is to make sure we can still pass in numbers

**Comment by mhvk** [comment]
p.s.  Really there should be more tests, but there are essentially none of `QuantityAttribute` to start with and I don't really have time to write a whole suite.

**Comment by bsipocz** [comment]
I'm sorry for the close/reopen, I was checking whether it would trigger circleCI to pick up the branch (as there was a system outage yesterday). 
@pllim - apparently it's not enough.

File: astropy/coordinates/builtin_frames/altaz.py [hunk 2]
```diff
@@ -52,9 +52,9 @@
     temperature : `~astropy.units.Quantity`
         The ground-level temperature as an `~astropy.units.Quantity` in
         deg C.  This is necessary for performing refraction corrections.
-    relative_humidity`` : numeric
-        The relative humidity as a number from 0 to 1.  This is necessary for
-        performing refraction corrections.
+    relative_humidity`` : `~astropy.units.Quantity`
```
[the remark this answers is not in the document]
Comment (mhvk) [reply astropy/coordinates/builtin_frames/altaz.py]: OK, done.

**Comment by mhvk** [comment]
Yes, a failure with how the test is written! Fixed now... (hurray for our testing framework, though! how often has it now shown me that I shouldn't trust my sense of something being obviously right...)

**Comment by bsipocz** [comment]
Thanks @mhvk!

**Pull request merged**